Doc/library/asyncio-task.rst

```diff
@@ -447,8 +448,11 @@ Running Tasks Concurrently
    Tasks/Futures to be cancelled.
 
    .. note::
-      A more modern way to create and run tasks concurrently and
-      wait for their completion is :class:`asyncio.TaskGroup`.
+      A new alternative to create and run tasks concurrently and
+      wait for their completion is :class:`asyncio.TaskGroup`. While *TaskGroup*
+      provides strong safety guarantees for scheduling a nesting of subtasks, *gather* comes in handy
+      for tasks that do not schedule subtasks and need to have their results consumed eagerly 
+      (i.e. when destructuring the result(s) into a tuple).
```

Comment:
I agree, my formulation didn't address this important point. Instead of committing your suggestion, I have tried to accomodate your suggestion differently, rewording this section of the documentation to better emphasize the difference and uses cases between _TaskGroup_ and _gather_. I don't want to stretch the text too much as this is not meant to be a tutorial, but I am all ears if this can be improved even more.

Reply:
I have made the requested changes; please review again.